### DISCUSSION TITLE
Partial object members not completed correctly in WIP sequence

### DISCUSSION THREAD

**Discussion by Willem-J-an** [microsoft/pylance-release#6846] [Enhancement]
Completion of partial object members does not work when editing a sequence, see below example.
```py
from enum import Enum


class X(Enum):
    my_enum_member = 0

(
    X. # << Completion here shows my_enum_member
    "something else"
)
(
    X.my_enu # << Completion here shows nothing
    "something else"
)
(
    X.my_enu, # << Completion before comma shows my_enum_member
    "something else"
)
```

I understand that if there's no comma written, the syntax is not valid, but when replacing something in a sequence it may occur that the new separator is not written until after writing the member. Additionally, it does work for the full completion, when there is no partial member name written yet. It seems inconsistent that it works for full completion, but not partial. A typical workflow for me would be to type until the completion is correct, and that workflow does not work in this case.

**Comment by erictraut** [comment]
The logic for completion suggestions within pyright is owned and maintained by the Microsoft pylance team.

Could someone from the pylance team please transfer this issue to the [pylance-release](https://github.com/microsoft/pylance-release) project and triage / prioritize it as you see fit? Thanks! @rchiodo, @debonte

**Comment by Waghabond** [comment]
This may be related to an issue I raised a while ago: #6716. Haven't heard back on that yet.

**Comment by rchiodo** [comment]
I don't think we'd special case this situation. There's a syntax error in this code which we'd have to try and guess what the intention was. Transferring to a discussion item for up votes.
